contracts/impl/TaskImpl.sol

```diff
@@ -291,9 +291,9 @@ library TaskImpl {
         // Verify the receipt
         verifyReceipt(state, _scAddr, _taskId, _stateDeltaHash, _gasUsed, msg.sender, _sig);
 
-        // Append the new state delta hash and set the contract's output hash
-        secretContract.stateDeltaHashes.push(_stateDeltaHash);
-        uint hashIndex = secretContract.outputHashes.push(_outputHash) - 1;
+        uint hashIndex = _stateDeltaHash != bytes32(0) ? secretContract.stateDeltaHashes.push(_stateDeltaHash) - 1 :
```

Comment:
I would name this variable here and in the event definition in a more informative way: `deltaIndex` or `deltaHashIndex`

Reply:
resolved: c0ddcde83e6a04514cd1e05611303c2d1dc1dbcf